frontend/src/metabase/dashboard/actions/parameters.ts

```diff
@@ -64,36 +93,40 @@ function updateParameters(dispatch, getState, parametersUpdater) {
   }
 }
 
-export const setEditingParameter = parameterId => dispatch => {
-  if (parameterId != null) {
-    dispatch(
-      setSidebar({
-        name: SIDEBAR_NAME.editParameter,
-        props: {
-          parameterId,
-        },
-      }),
-    );
-  } else {
-    dispatch(closeSidebar());
-  }
-};
+export const setEditingParameter =
+  (parameterId: ParameterId) => (dispatch: Dispatch) => {
+    if (parameterId != null) {
+      dispatch(
+        setSidebar({
+          name: SIDEBAR_NAME.editParameter,
+          props: {
+            parameterId,
+          },
+        }),
+      );
+    } else {
+      dispatch(closeSidebar());
+    }
+  };
 
 export const ADD_PARAMETER = "metabase/dashboard/ADD_PARAMETER";
 export const addParameter = createThunkAction(
   ADD_PARAMETER,
-  parameterOption => (dispatch, getState) => {
-    let parameter;
+  (option: ParameterMappingOptions) => (dispatch, getState) => {
+    let newId: ParameterId;
```

Comment:
Maybe set to `undefined` here and in `dispatch` call `checkNotNull` or similar to make an explicit check?

Reply:
This is actually a good idea, I have initialized it with `undefined`, which helped the compiler understand the intention without a warning.

I am using an explicit check instead of `checkNotNull` because the latter throws exceptions.